users/engler/fairseq/training.py

```diff
@@ -109,20 +145,28 @@ def create_files(self):
         util.create_executable("fairseq.sh", self._get_run_cmd())
 
     def run(self):
-        sp.check_call(self._get_run_cmd())
+        my_env = os.environ
+        if self.fairseq_root is not None:
+            my_env["PYTHONPATH"] = self.fairseq_root
+        sp.check_call(self._get_run_cmd(), env=my_env)
 
     def _get_run_cmd(self):
         run_cmd = [
-            tk.uncached_path(self.fairseq_hydra_exe),
             "--config-dir",
             os.path.dirname(self.out_fairseq_hydra_yaml.get_path()),
             "--config-name",
             os.path.basename(self.out_fairseq_hydra_yaml.get_path()),
         ]
         run_cmd += self.command_line_args
-        run_cmd += ["checkpoint.save_dir=" + str(self.out_checkpoint_dir)]
+        run_cmd += ["checkpoint.save_dir=" + self.out_checkpoint_dir.get_path()]
         run_cmd += ["checkpoint.save_interval=" + str(self.save_interval)]
         run_cmd += ["optimization.max_epoch=" + str(self.max_epoch)]
-        if self.fairseq_python_exe is not None:
+        if self.fairseq_root is not None:
+            sys.path.insert(0, self.fairseq_root)
+            hydra_train_entry = self.fairseq_root + "fairseq_cli/hydra_train.py"
+            run_cmd.insert(0, tk.uncached_path(self.fairseq_python_exe))
+            run_cmd.insert(1, tk.uncached_path(hydra_train_entry))
```

Comment:
I think `run_cmd = [tk.uncached_path(...), tk.uncached_path(...)] + run_cmd` is easier to read, but that's a matter of taste.